share/completion/git-restore

```diff
@@ -0,0 +1,48 @@
+# (C) 2024 unrealapex
+
+# Completion script for the "git-restore" command.
+
+function completion/git-restore {
+        WORDS=(git restore "${WORDS[2,-1]}")
+        command -f completion//reexecute
+}
+
+function completion/git::restore:arg {
+
+        OPTIONS=( #>#
+        "s --source=; restore the working tree files with the content from the given tree"
+        "p --patch; interactively select hunks in the difference between the restore source and the restore location"
+        "W --worktree; specify the restore location"
+        "S --staged; specify the restore location"
+        "q --quiet; quiet, supress feedback messages"
+        "--progress; display progress status"
+        "--no-progress; do not display progress status"
```

Comment:
This flag's description was edited for brevity. Should the descriptions be verbatim from the man page?

Reply:
I usually rewrite descriptions so that they fit in small terminal screens.